test/e2e/set/scyllacluster/scyllacluster_cleanup.go

```diff
@@ -0,0 +1,169 @@
+		o.Expect(cleanupJobsCreated).To(o.ContainElements(
+			o.Satisfy(nodeJobMatcher(naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 0))),
+			o.Satisfy(nodeJobMatcher(naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 1))),
+		))
+
+		framework.By("Scaling down the ScyllaCluster to 2 replicas")
+
+		jobObserver = utils.ObserveObjects[*batchv1.Job](jobListWatcher)
+		jobObserver.Start(ctx)
+
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(
+			ctx,
+			sc.Name,
+			types.JSONPatchType,
+			[]byte(`[{"op": "replace", "path": "/spec/datacenter/racks/0/members", "value": 2}]`),
+			metav1.PatchOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(sc.Spec.Datacenter.Racks).To(o.HaveLen(1))
+		o.Expect(sc.Spec.Datacenter.Racks[0].Members).To(o.BeEquivalentTo(2))
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx3, waitCtx3Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx3Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx3, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+
+		framework.By("Validate cleanup jobs were created for all nodes")
+		jobEvents, err = jobObserver.Stop()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Expect(jobEvents).ToNot(o.BeEmpty())
+
+		cleanupJobsCreated = helpers.Filter(jobEvents, func(e utils.ObserverEvent[*batchv1.Job]) bool {
+			return e.Action == watch.Added && e.Obj.Labels[naming.NodeJobTypeLabel] == naming.JobTypeCleanup
+		})
+
+		o.Expect(cleanupJobsCreated).To(o.HaveLen(2))
+		o.Expect(cleanupJobsCreated).To(o.ContainElements(
+			o.Satisfy(nodeJobMatcher(naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 0))),
+			o.Satisfy(nodeJobMatcher(naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 1))),
+		))
+
+		firstNodeCleanupJobCompleted, eventFound := helpers.Find(jobEvents, func(e utils.ObserverEvent[*batchv1.Job]) bool {
+			return e.Obj.Status.CompletionTime != nil &&
+				e.Obj.Labels[naming.NodeJobTypeLabel] == naming.JobTypeCleanup &&
+				e.Obj.Labels[naming.NodeJobLabel] == naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 0)
+		})
+		o.Expect(eventFound).To(o.BeTrue())
+
+		secondNodeCleanupJobCreated, eventFound := helpers.Find(jobEvents, func(e utils.ObserverEvent[*batchv1.Job]) bool {
+			return e.Action == watch.Added &&
+				e.Obj.Labels[naming.NodeJobTypeLabel] == naming.JobTypeCleanup &&
+				e.Obj.Labels[naming.NodeJobLabel] == naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 1)
+		})
+		o.Expect(eventFound).To(o.BeTrue())
+
+		o.Expect(firstNodeCleanupJobCompleted.Obj.Status.CompletionTime).ToNot(o.BeNil())
+		o.Expect(firstNodeCleanupJobCompleted.Obj.Status.CompletionTime.Time.Compare(secondNodeCleanupJobCreated.Obj.CreationTimestamp.Time)).To(o.BeNumerically("<=", 0))
+	})
```

Comment:
Shouldn't we also wait for the second job to complete successfully?

Reply:
Jobs must complete before ScyllaCluster reports Progressing=False, this check verifies whether jobs are created sequentially. But this limitation was lifted since core approved that cleanup no longer impacts cluster that much to influence user traffic, so jobs are run in parallel.